lib/re1.5/compilecode.c

```diff
@@ -89,19 +94,21 @@ static const char *_compilecode(const char *re, ByteProg *prog, int sizecode)
         case '(': {
             term = PC;
             int sub = 0;
-            int capture = re[1] != '?' || re[2] != ':';
+            int capture = re_top - re > 2 && (re[1] != '?' || re[2] != ':');
```

Comment:
To be fair, some of the extra size in the null-content commit is the extra length safety such as this line, preventing potential buffer overflows in the existing code. 

That being said many of the similar / new checks above like `if (re >= re_top) return NULL;` weren't needed in the original code becase it was guaranteed that there's always the "extra" NULL byte on the end of teh "valid" string data so the code can get away with a `re++; *re == char` once per NULL check.